sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -796,71 +796,75 @@ size_t gcoap_req_send2(const uint8_t *buf, size_t len,
                        gcoap_resp_handler_t resp_handler)
 {
     gcoap_request_memo_t *memo = NULL;
+    unsigned msg_type  = (*buf & 0x30) >> 4;
+
     assert(remote != NULL);
 
     /* Find empty slot in list of open requests. */
```

Comment:
Move this comment inside the conditional? I found it confusing to leave here. OTOH, it might be good to add a new comment here, like "Allocate memo if necessary"